server/server.js

```diff
@@ -4,65 +4,69 @@ var url = require('url');
 // Take the listening port as argument
 var port = (process.argv.length > 2 ? process.argv[2] : 8100);
 
-var handleApiRequest = function (request, response, apiHandler) {
+var handleApiRequest = function (request, response) {
 
-  console.log("Serving API");
+    console.log("Serving API");
 
-  // Employing very forgiving CORS, so this API-endpoint can be called from another origin;
-  // E.g. when running multiple instances of this server that makes API-calls between each other.
-  var responseHeaders = {
-    'Content-Type': 'application/json',
-    'Access-Control-Allow-Origin': '*',
-    'Access-Control-Allow-Headers': 'Authorization'
-  };
+    var responseHeaders = {
+        'Content-Type': 'application/json',
+    };
 
-  // Making the naive assumption that any OPTIONS request is the user-agent making a cross-site preflighted request.
-  if (request.method === 'OPTIONS') {
-    console.log("Accepting probable preflight request");
+    var origin = request.headers["origin"];
 
+    // Making the naive assumption that any OPTIONS request is the user-agent making a cross-site preflighted request.
```

Comment:
this actually isn't naive or assumed. It's checked on line 21. If `origin` is present, this is a CORS OPTIONS request -- i.e., preflight. Comment can be removed.

Reply:
done